src/ProjectTemplates/TestInfrastructure/PrepareForTest.targets

```diff
@@ -61,22 +65,15 @@
 
     <Delete Files="@(_ExistingFilesFromLastRun)" ContinueOnError="true" />
 
-    <Removedir Directories="$(TestTemplateCreationFolder)" Condition="Exists('$(TestTemplateCreationFolder)')" ContinueOnError="true">
+    <RemoveDir Directories="$(TestTemplateCreationFolder)" Condition="Exists('$(TestTemplateCreationFolder)')" ContinueOnError="true">
       <Output TaskParameter="RemovedDirectories" ItemName="_CleanedUpDirectories" />
-    </Removedir>
-    <Removedir Directories="$(TestPackageRestorePath)" Condition="Exists('$(TestPackageRestorePath)')" ContinueOnError="true">
+    </RemoveDir>
+    <RemoveDir Directories="$(TestPackageRestorePath)" Condition="Exists('$(TestPackageRestorePath)')" ContinueOnError="true">
       <Output TaskParameter="RemovedDirectories" ItemName="_CleanedUpDirectories" />
-    </Removedir>
+    </RemoveDir>
 
     <Message Importance="high" Text="Removed directory %(_CleanedUpDirectories.Identity)" />
 
-    <MakeDir Directories="$(TestTemplateCreationFolder)">
-      <Output TaskParameter="DirectoriesCreated" ItemName="_CreatedDirectories" />
-    </MakeDir>
-    <MakeDir Directories="$(TestPackageRestorePath)">
-      <Output TaskParameter="DirectoriesCreated" ItemName="_CreatedDirectories" />
-    </MakeDir>
```

Comment:
Restore operations automatically create target folders, as does the `GenerateFileFromTemplate` task. No need for these extra tasks.